Update color palette for accessibility

kartikcho opened the pull request:
The current color palette is too similar for people with color deficiency people to differentiate. 

One of the comments from the user feedback: 
`"...was slightly confused why one was red and another blue and thought they indicate bad/good perf"`

This PR updates color choices to fix these where the following has been changed,

- Red (commit events) -> Yellow
- Blue (user timing marks) -> Grey
- Purple (measures) -> slightly lighter purple on hover

![image](https://user-images.githubusercontent.com/48270786/89286012-13cae780-d66f-11ea-99c1-1a674c987fac.png)

Resolves #55 

**Edit:** Update snapshot due to a failing test unrelated to the changes made in this PR.
**Edit 2:** Reverted snapshot update which was failing on my machine only (still is, caused by a switched sequence of [these lines](https://github.com/MLH-Fellowship/scheduling-profiler-prototype/blob/06606033fd96a7c46392e4236339ecf54af0313f/src/util/__tests__/preprocessData-test.js#L330-L336). Weird.)

kartikcho committed:
Update color palette to be more accessible

kartikcho committed:
Update color palette to be more accessible

kartikcho committed:
Update jest snapshot

kartikcho committed:
Update jest snapshot

kartikcho committed:
Revert "Update jest snapshot"

This reverts commit ae48474bf44adbd0c82aae404fbcd7df0a362a2c.

kartikcho committed:
Revert "Update jest snapshot"

This reverts commit ae48474bf44adbd0c82aae404fbcd7df0a362a2c.

taneliang commented:
As a red-green color deficient person, here's my feedback:

* The commit phase yellows: the unhovered and selected yellows look identical, and the hover yellow looks almost identical to the other 2 yellows
* The commit phase yellows don't pop out of the gray background well enough -- they kinda blend together if the commit phase is short.
* The unhovered render blue and the layout effect blue are an identical sky blue (this is also an issue on `master`)
* For layout/passive effects, the selected and hovered blues are too similar (on `master` hovered is clearly a lot darker)

This yellow text is also too bright imo and is hard to read
![image](https://user-images.githubusercontent.com/12784593/89294236-cca21e00-d691-11ea-8d29-efc8e9925cf3.png)

Also, what did Brian say again about the commit phase reds? I remember he said something about the reds being intentional because if they're too long it's an opportunity for improvement. I also suspect that the red colors for the cascading suspends are intentional. I think the new yellows don't communicate this as well as the reds. Would be great if we could clarify both with him (or if you have then that's great)

kartikcho commented:
>The commit phase yellows: the unhovered and selected yellows look identical, and the hover yellow looks almost identical to the other 2 yellows

I'll update them to be more accessibly distinct.

>The unhovered render blue and the layout effect blue are an identical sky blue (this is also an issue on master)

I think they're distinct enough as is but I'll increase all hover colors to be more noticeable. 

>For layout/passive effects, the selected and hovered blues are too similar (on master hovered is clearly a lot darker)

They're purple and I'll make the hover purple more distinguishable.

>What did Brian say again about the commit phase reds?

Yellow would be a better choice over red as feedback shows that it's confusing users to think it's an error or something. Also, afaik React profiler doesn't use red either so when integrating with the Devtools, this could be another confusing element to the users.

I'll ping @bvaughn to provide clarity on this.

taneliang committed:
Improve Firefox scrolling performance (#114)

taneliang committed:
Codemod pan/hover interactions -> mouse interactions (#115)

taneliang committed:
Listen to mouse move events when dragging offscreen (#116)

taneliang committed:
Upgrade dependencies, including Speedscope fork with upstreamed fix (#118)

bvaughn commented:
> > What did Brian say again about the commit phase reds?
>
> Yellow would be a better choice over red as feedback shows that it's confusing users to think it's an error or something. Also, afaik React profiler doesn't use red either so when integrating with the Devtools, this could be another confusing element to the users.

React profiler avoids the use of red because of [red-green colorblindness](https://www.color-blindness.com/deuteranopia-red-green-color-blindness/) concerns. I did some research at the time and found the color gradient DevTools *does* use (a blue to yellow) was more accessible.

That being said, this profiler is a bit of a different case. Its colors are not a gradient, and it has more distinct concepts needing colors (although admittedly we can also use shapes and placement to differentiate here).

Since we have an easily changeable color config so that we can iterate on these, I'm not overly concerned at this point. In the future at some point maybe we'll even make it configurable (e.g. a high contrast mode).

taneliang committed:
Normalize wheel behavior across browsers (#117)

taneliang committed:
Reorganize layout and canvas folders (#120)

taneliang committed:
Reimplement zoom to batch (#123)

taneliang committed:
Fix tooltip (#121)

* Minor clean up of EventTooltip.js

* Fix root cause of #47 and improve tooltip styling

Tooltips would go offscreen because of the way `top` is defined. This
commit fixes that root cause by changing CanvasPage to position fixed so
that tooltips will be positioned relative to the DOM rather than the
CanvasPage div.

Undoes some of the changes made in #78 as this commit fixes the root
cause of the bug.

* Disable component name truncation

Although this was Brian's recommendation (see #67), I don't think
component names will be very long because the source code for those will
be really annoying.

* Set mouse location on every interaction to fix #87

* Introduce 768-char component name length limit

Covers the unlikely edge case that a component name is unreasonably
long.

kartikcho committed:
Update color palette

kartikcho committed:
Merge branch 'a11y-color-update' of https://github.com/MLH-Fellowship/scheduling-profiler-prototype into a11y-color-update
